test/functional/lua/system_spec.lua

```diff
@@ -75,14 +75,38 @@ describe('vim.system', function()
       it('supports timeout', function()
         eq({
           code = 124,
-          signal = 15,
+          signal = 15, -- SIGTERM
           stdout = '',
           stderr = '',
         }, system({ 'sleep', '10' }, { timeout = 1000 }))
       end)
     end)
   end
 
+  it('supports timeout = 0', function()
+    local ret
+
+    ret = exec_lua [[
+      return vim.system({ 'sleep', '1' }, { timeout = 0 }):wait()
+    ]] --[[ @as vim.SystemCompleted ]]
+    eq({
+      code = 124,
+      signal = 15, -- SIGTERM
+      stdout = '',
+      stderr = '',
+    }, ret)
+
+    ret = exec_lua [[
+      return vim.system({ 'sleep', '1' }):wait(0)
+    ]] --[[ @as vim.SystemCompleted ]]
+    eq({
+      code = 124,
+      signal = 15, -- SIGTERM
+      stdout = '',
+      stderr = '',
+    }, ret)
+  end)
+
```

Comment:
The docs say `timeout` should be `>= 1`, so what is the purpose of having a test for `timeout = 0`? A zero timeout is "unspecified" behavior according to our docs, so do we need a test?

Eventually (post 0.10) we may want to add some well-defined behavior for `timeout = 0` in which case this test would test for that specific behavior.

Reply:
+1 Removed the test case and left some comment about the 'underspecified' behavior in the implementation.